lightning/src/routing/router.rs

```diff
@@ -493,12 +493,22 @@ pub struct Path {
 impl Path {
 	/// Gets the fees for a given path, excluding any excess paid to the recipient.
 	pub fn fee_msat(&self) -> u64 {
```

Comment:
I think this method a little hard to read atm. Looks like it can be simplified:
```rust
	pub fn fee_msat(&self) -> u64 {
		let num_unblinded_hops = self.hops.len() + self.trampoline_hops.len();
		let unblinded_hop_fees_iter = self.hops.iter().map(|hop| hop.fee_msat)
			.chain(self.trampoline_hops.iter().map(|hop| hop.fee_msat));
		match &self.blinded_tail {
			Some(_) => unblinded_hop_fees_iter.sum::<u64>(),
			None => {
				// Do not count last hop of each path since that's the full value of the payment
				unblinded_hop_fees_iter.take(num_unblinded_hops - 1).sum()
			}
		}
	}